### PR TITLE
Migrate configurations-generator

### PR DESCRIPTION
**Pull request by Demonsthere**
**Description**
Clean PR
- Update Jenkinsfile
- Create before-commit
- Fix tests

**Related issue(s)**

Fixes https://github.com/kyma-project/kyma/pull/1998
Implements https://github.com/kyma-project/test-infra/issues/224

### CODE REVIEW

File: components/configurations-generator/Jenkinsfile [hunk 1]
```diff
@@ -54,19 +54,19 @@ podTemplate(label: label,
                         }
 
                         stage("install dependencies $application") {
-                            execute("make resolve", workDir, componentDir)
+                            execute("dep ensure --vendor-only -v", workDir, componentDir)
                         }
 
                         stage("code quality $application") {
-                            execute("make validate", workDir, componentDir)
+                            execute("gometalinter --vendor --deadline=2m --disable-all --enable=vet --skip=\$(go list ./... | grep \"/fake\") ./...", workDir, componentDir)
```
Comment (kubadz) [inline components/configurations-generator/Jenkinsfile]: we don't need to skip packages containing "/fake" as there are no such packages in this component

File: components/configurations-generator/before-commit.sh [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+##
+# GO BUILD
+##
+buildEnv=""
+if [ "$1" == "$CI_FLAG" ]; then
+	# build binary statically
+	buildEnv="env CGO_ENABLED=0"
+fi
+
+${buildEnv} go build -o generator ./cmd/generator/
```
Comment (Tomasz-Smelcerz-SAP) [inline components/configurations-generator/before-commit.sh]: Have you ever run it?
Output file path (relative to project dir) is: ./generator. In Dockerfile you build image from file ./bin/app

File: components/configurations-generator/before-commit.sh [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+echo "? go test"
+go test ./...
+# Check if tests passed
+if [ $? != 0 ]; then
+	echo -e "${RED}✗ go test\n${NC}"
+	exit 1
+else echo -e "${GREEN}√ go test${NC}"
+fi
+
+goFilesToCheck=$(find . -type f -name "*.go" | egrep -v "\/vendor\/|_*/automock/|_*/testdata/|/pkg\/|_*export_test.go")
```
Comment (kubadz) [inline components/configurations-generator/before-commit.sh]: why do you skip go files in the `/pkg` directory? I would say that we should format them, as in most cases they contain code written by us.

File: components/configurations-generator/before-commit.sh (hunk 2 above)
Comment (Demonsthere) [reply components/configurations-generator/before-commit.sh]: Good catch, fixing!

File: components/configurations-generator/Jenkinsfile (hunk 1 above)
Comment (Demonsthere) [reply components/configurations-generator/Jenkinsfile]: Fixing!

File: components/configurations-generator/before-commit.sh (hunk 3 above)
Comment (Demonsthere) [reply components/configurations-generator/before-commit.sh]: Good call, fixed.

**Pull request merged**